apps/meteor/app/livechat/server/lib/FeedbackSurveyMonitor.ts

```diff
@@ -0,0 +1,85 @@
+import { SyncedCron } from 'meteor/littledata:synced-cron';
+
+import LivechatRooms from '../../../models/server/models/LivechatRooms';
+import { fetch } from '../../../../server/lib/http/fetch';
+import { settings } from '../../../settings/server';
+
+export class FeedbackSurveyMonitor {
+	private _started: boolean;
+
+	private _name: string;
+
+	private _dialogflowIncomingUrl: string;
+
+	constructor() {
+		this._started = false;
+		this._name = 'Feedback Survey Monitor';
+		this._dialogflowIncomingUrl = settings.get('Livechat_dialogflow_incoming_url') || '';
+	}
+
+	start() {
+		this._startMonitoring();
+	}
+
+	stop(): void {
+		if (!this.isRunning()) {
+			return;
+		}
+
+		SyncedCron.remove(this._name);
+
+		this._started = false;
+	}
+
+	isRunning() {
+		return this._started;
+	}
+
+	_startMonitoring() {
+		if (this.isRunning()) {
+			return;
+		}
+		const everyMinute = '* * * * *';
+		SyncedCron.add({
+			name: this._name,
+			schedule: (parser) => parser.cron(everyMinute),
+			job: () => {
+				this._handleFeedbackSurveyRooms();
+			},
+		});
+		this._started = true;
+	}
+
+	async _handleFeedbackSurveyRooms() {
+		if (!this._dialogflowIncomingUrl || this._dialogflowIncomingUrl === '') {
+			return;
+		}
+
+		const rooms = await LivechatRooms.findOpenWithFeedbackSurveyCustomField();
+		rooms.forEach(async (room: any) => {
+			const surveyTimer = room.customFields?.feedbackSurvey;
+			if (surveyTimer && room.lastMessage && room.lastMessage?.u?._id !== room?.v?._id) {
+				const currentTime = new Date();
+				const lastMessageTime = new Date(room.lastMessage.ts);
+				if (lastMessageTime.getTime() + surveyTimer * 1000 <= currentTime.getTime()) {
```

Comment:
@bhardwajaditya can we a log point for when we are triggering the feedback survey for a particular room please?  Need to include the RoomId.  Thanks.